### DISCUSSION TITLE
Why Orange is slower than sklearn

### DISCUSSION THREAD

**Discussion by gengyabc** [biolab/orange3#6006] [General]
I tried to work on this kaggle competition [Alice - logistic regression baseline
](https://www.kaggle.com/code/kashnitsky/alice-logistic-regression-baseline/notebook) using Orange. But I find Orange speed is too slow, especially the 'test and score' made me wait for 10+ minutes(I used that time cooked my lunch...). I think Orange calculation is done by sklearn, but using this kernel only takes me several seconds. Why there is such a big speed difference.

**Comment by janezd** [comment]
It depends upon what you were doing, how you were testing. Orange indeed uses sklearn, but sometimes it must add something to prepare the data, reinterpret sklearns results ... and sometime its just inefficient. We'd need to know more to be able to say what's going on.

**Comment by gengyabc** [comment]
Thank you for your reply. This is the screen shot
![image](https://user-images.githubusercontent.com/12781437/172079585-6fed5969-6411-4c8d-8f57-092daa8c3256.png)
![image](https://user-images.githubusercontent.com/12781437/172079599-796d3e25-29fa-4fa3-a3de-a90ed2f38a12.png)


What I am doing is: first load the data(This is slower than pd.csv_read, maybe the data need to be prepared first? ), and I changed 'siteX' features to text in order to use the bag of words method. And last is the test and score, which is extremely slow.

**Reply by gengyabc** [reply]
finally the calculation is finished, but this time there is something wrong:
￼ Logistic Regression failed with error: (1 warning)
￼ Logistic Regression failed with error:
DomainTransformationError: Variables 'target' in the model is incompatible with the variable of the same name in the data.

￼ Some scores could not be computed.

Previously, I got no error, but just slow speed.

This is my file:
[alice.zip](https://github.com/biolab/orange3/files/8841271/alice.zip)
